From fd357852d0714c87a6383578f277a02155e213b2 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 7 May 2020 20:41:19 +0530
Subject: [PATCH 0497/1921] octeontx2-af: Do not change maximum NPC interfaces

As per the software design for loading profiles from
firmware, changing the maximum interfaces macro causes
incompatibility with existing profile binaries.
Hence the macro is reverted.

Change-Id: I2cdd28ed50f6692e0bad02b734f8c5cdee5a0ad1
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/28053
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 1aa7b2bcea10..e23b571c961b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -1309,7 +1309,7 @@ struct npc_get_kex_cfg_rsp {
 	struct mbox_msghdr hdr;
 	u64 rx_keyx_cfg;   /* NPC_AF_INTF(0)_KEX_CFG */
 	u64 tx_keyx_cfg;   /* NPC_AF_INTF(1)_KEX_CFG */
-#define NPC_MAX_INTF	4
+#define NPC_MAX_INTF	2
 #define NPC_MAX_LID	8
 #define NPC_MAX_LT	16
 #define NPC_MAX_LD	2
-- 
2.31.1

